### PR TITLE
[BEAM-10018] Fix timestamps in two windowing Python katas

### PR DESCRIPTION
**Pull request by iht**
The two Python katas about windowing fail because the timestamps for the elements are calculated based on the local timezone, and my timezone does not match the timezone hardcoded in the tests, and because parsing from strings using `fromisoformat` was failing.

In the first Kata, the timestamp was calculated from time objects, and converted to a timestamp in the local timezone. Thus, the results of the test depended on the configuration of the local timezone in the running system.

The tests were hardcoded with a timezone different to mine, and thus I always failed to pass this Kata. The changes in this commit change the type in `Event` to be a `datetime`, the timestamps are set in UTC, and the output in the tests is hardcoded in UTC too. This should ensure that the kata works regardless the timezone configured in the system running the kata.

In the second Kata, the code was failing with the following error:

```AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'```

I changed the timestamps to be set using the `datetime` constructor, rather than parsing from strings using `fromisoformat`.

Both katas now pass with the examples hardcoded in the corresponding tests.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by iht** [comment]
R: @henryken

File: learning/katas/python/Windowing/Adding Timestamp/ParDo/task.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 import datetime
 import time
```
[the remark this answers is not in the document]
Comment (iht) [reply learning/katas/python/Windowing/Adding Timestamp/ParDo/task.py]: Done

File: learning/katas/python/Windowing/Adding Timestamp/ParDo/task-info.yaml [hunk 2]
```diff
@@ -22,10 +22,10 @@ files:
 - name: task.py
   visible: true
   placeholders:
-  - offset: 1211
-    length: 163
+  - offset: 1243
```
[the remark this answers is not in the document]
Comment (iht) [reply learning/katas/python/Windowing/Adding Timestamp/ParDo/task-info.yaml]: I have updated the placeholder values, and I have checked the course using Course preview. I think the placeholders are now correct.

File: learning/katas/python/Windowing/Adding Timestamp/ParDo/task.py [hunk 3]
```diff
@@ -24,30 +25,30 @@
 
 
 class Event:
-    def __init__(self, id, event, date):
+    def __init__(self, id, event, timestamp):
         self.id = id
         self.event = event
-        self.date = date
+        self.timestamp = timestamp
```
[the remark this answers is not in the document]
Comment (iht) [reply learning/katas/python/Windowing/Adding Timestamp/ParDo/task.py:30]: Fixed

File: learning/katas/python/Windowing/Fixed Time Window/Fixed Time Window/task-info.yaml [hunk 4]
```diff
@@ -22,7 +22,7 @@ files:
 - name: task.py
   visible: true
   placeholders:
-  - offset: 2074
+  - offset: 2067
```
[the remark this answers is not in the document]
Comment (iht) [reply learning/katas/python/Windowing/Fixed Time Window/Fixed Time Window/task-info.yaml:25]: I am checking the boundaries of the placeholder in PyCharm, and I see it correctly
image
Maybe you were checking the previous placeholder box? I see that the box is starting 7 positions to the right (2074, after the `Wi`), and the placeholder I am submitting is starting 7 positions to the left (2067).

I have removed all the Placeholders and created them again by selecting the answer in the code, and using right click "Add answer placeholder", and the bounding boxes are the same I have committed (same values in the section `placeholders` in `task-info.yaml`). I am not sure if I am doing something wrong though. It is the first time I generate Edu content for PyCharm.

**Comment by henryken** [comment]
Thanks for the changes @iht. LGTM now.
@pabloem, can you help to merge and close this issue?

**Comment by pabloem** [comment]
making sure license tests pass

**Comment by pabloem** [comment]
thanks everyone!

**Comment by iht** [comment]
Thank you!

**Pull request merged**